[PLAY-54]: added network-graph component

AntonNos opened the pull request:
Added `network-graph` component as a own npm package.
<img width="1728" alt="Screenshot 2023-05-25 at 21 28 42" src="https://github.com/dechegaray/react-graph-network/assets/33761040/0026899c-a47e-425c-a571-84cadcbf083a">

Christopher-Gustavsson commented on src/.DS_Store:
Lets add `.DS_Store` to the `.gitignore` file so we avoid this file.

AntonNos replied:
Oh, sorry, I forgot about this one.
Removed and added it ti `.gitignore`

Christopher-Gustavsson commented on src/index.test.tsx:
We can probably remove this test. I'm guessing this was just to see if the configs we working or something like that.

AntonNos replied:
Yes, right. I added it to check that config works well. Removed

AntonNos replied:
Hmm, I pushed update without test and build fails. Probably we need to add at least 1 test

AntonNos replied:
I added another test file, which includes simple test for navigation controls

Christopher-Gustavsson reviewed (changes requested):
Awesome work. Just left some minor comments nothing big. 
Other than those I think this is looking pretty good. Good idea to include an example!

AntonNos commented:
> Awesome work. Just left some minor comments nothing big. Other than those I think this is looking pretty good. Good idea to include an example!

Thanks! Comments were fixed.

dechegaray commented on package.json:
```diff
@@ -1,49 +1,98 @@
+    "cross-env": "^7.0.2",
     "css-loader": "^6.7.3",
-    "eslint": "^8.39.0",
+    "d3-drag": "^3.0.0",
```
let's move these d3-packages to `dependencies`

AntonNos replied:
Done

dechegaray commented on package.json:
```diff
@@ -1,49 +1,98 @@
+    "@types/react-dom": "^18.2.4",
+    "@typescript-eslint/eslint-plugin": "^5.59.7",
+    "@typescript-eslint/parser": "^5.59.7",
+    "@visx/responsive": "^3.0.0",
```
visx/responsive and zoom should also go to `dependencies`

AntonNos replied:
Done

dechegaray reviewed (changes requested):
A couple of important changes we need to have the package as clean as possible 👍

AntonNos commented on dist/index.js:
Done

AntonNos commented on example/src/App.tsx:
Done

AntonNos commented on package.json:
```diff
@@ -1,49 +1,98 @@
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-scripts": "^5.0.1",
     "typescript": "^5.0.4"
```
Done

AntonNos commented on package.json:
```diff
@@ -1,49 +1,98 @@
-    "webpack": "^5.81.0",
-    "webpack-cli": "^5.0.2",
-    "webpack-dev-server": "^4.13.3"
+    "type-check": "^0.4.0"
```
It is used for github action - `"typecheck": "tsc --noEmit --declaration false"`

AntonNos commented on package.json:
```diff
@@ -1,49 +1,98 @@
     "jest-junit": "^16.0.0",
-    "prettier": "^2.8.8",
+    "microbundle-crl": "^0.13.10",
+    "npm-run-all": "^4.1.5",
```
I removed this one - `"npm-run-all": "^4.1.5"`, but `microbundle-crl` is used for `start` and `build` scripts

AntonNos commented on package.json:
```diff
@@ -1,49 +1,98 @@
     "jest": "^29.5.0",
     "jest-canvas-mock": "^2.5.0",
     "jest-environment-jsdom": "^29.5.0",
     "jest-junit": "^16.0.0",
```
Done

AntonNos commented on README.md:
````diff
@@ -0,0 +1,30 @@
+## Install
+
+```bash
+npm install --save network-graph
````
Done

AntonNos commented:
> A couple of important changes we need to have the package as clean as possible 👍

Thanks for comments, I double checked packages and removed unused. Other comments were also fixed.
